src/cargo/core/shell.rs

```diff
@@ -76,12 +114,35 @@ impl MultiShell {
         self.err().say(message, YELLOW)
     }
 
+    pub fn set_verbosity(&mut self, verbose: bool, quiet: bool) -> CargoResult<()> {
+        self.verbosity = match (verbose, quiet) {
+            (true, true) => {
+                return Err(Box::new(VerbosityError));
+            },
+            (true, _) => {
+                Verbose
+            },
+            (_, true) => {
+                Quiet
+            }
+            _ => {
+                Normal
+            }
+        };
```

Comment:
This would be a little more concise and exhaustive as:

```rust
self.verbosity = match (verbose, quiet) {
    (true, true) => return Err(human("...")),
    (true, false) => Verbose,
    (false, true) => Quiet,
    (false, false) => Normal,
};